internal/interconnections/retrieve.go

```diff
@@ -0,0 +1,91 @@
+//
+// Permission is hereby granted, free of charge, to any person obtaining a copy
+// of this software and associated documentation files (the "Software"), to deal
+// in the Software without restriction, including without limitation the rights
+// to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+// copies of the Software, and to permit persons to whom the Software is
+// furnished to do so, subject to the following conditions:
+//
+// The above copyright notice and this permission notice shall be included in
+// all copies or substantial portions of the Software.
+//
+// THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+// IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+// FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+// AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+// LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+// OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN
+// THE SOFTWARE.
+
+package interconnections
+
+import (
+	"context"
+	"fmt"
+
+	metal "github.com/equinix-labs/metal-go/metal/v1"
+	pager "github.com/equinix/metal-cli/internal/pagination"
+	"github.com/spf13/cobra"
+)
+
+func (c *Client) Retrieve() *cobra.Command {
+	// metrosCmd represents the metros command
+	var projectID, organizationID, connID string
+	retrieveInterconnectionsCmd := &cobra.Command{
+		Use:     "get",
+		Aliases: []string{"list"},
+		Short:   "Retrieves interconnections for the current user, an organization, a project or the details of a specific interconnection.",
+		Long:    "Retrieves interconnections for the current user, an organization, a project or the details of a specific interconnection.",
+		Example: `  # Retrieve all interconnections of a current user::
+  
+  # Retrieve the details of a specific interconnection:
+  metal interconnections get -i e9a969b3-8911-4667-9d99-57cd3dd4ef6f
+
+  # Retrieve all the interconnection of an organization:
+  metal interconnections get -O c079178c-9557-48f2-9ce7-cfb927b81928
+
+  # Retrieve all interconnection of a project:
+  metal interconnections get -p 1867ee8f-6a11-470a-9505-952d6a324040 `,
+
+		RunE: func(cmd *cobra.Command, args []string) error {
+			cmd.SilenceUsage = true
+			var interConnections []metal.Interconnection
+			var proerr error
+			inc := []string{}
+			exc := []string{}
+			header := []string{"ID", "Name", "Code", "Type", "Created"}
+
+			if projectID != "" {
+				interConnections, proerr = pager.GetAllProjectInterconnections(c.Service, projectID, inc, exc)
+				if proerr != nil {
+					return fmt.Errorf("could not list Project Interconnections: %w", proerr)
+				}
+			} else if organizationID != "" {
+				interConnectionsList, _, err := c.Service.OrganizationListInterconnections(context.Background(), organizationID).Include(inc).Exclude(exc).Execute()
```

Comment:
Pagination is being handled for project interconnections now, but it looks like there is also pagination data available for OrganizationListInterconnections